packages/frontend/components/hooks/useAutoConnect.ts

```diff
@@ -0,0 +1,92 @@
+import { useEffect } from "react";
+import { Connector, useAccount, useConnect } from "wagmi";
+import { useEffectOnce, useLocalStorage } from "usehooks-ts";
+import { burnerWalletId, defaultBurnerChainId } from "~~/web3/wagmi-burner";
+import { useBurnerWallet } from "~~/components/hooks/useBurnerWallet";
+
+// const supportedRpcUrls = [
+//   chain.localhost.rpcUrls.default,
+//   chain.localhost.rpcUrls.default.replace("127.0.0.1", "localhost"),
+//   chain.hardhat.rpcUrls.default,
+//   chain.localhost.rpcUrls.default.replace("127.0.0.1", "localhost"),
+// ];
+
+export type TAutoConnect = {
+  /**
+   * Enable the burner wallet.  If this is disabled, burner wallet is entierly disabled
+   */
+  enableBurnerWallet: boolean;
+  /**
+   * Auto connect:
+   * 1. If the user was connected into a wallet before, on page reload reconnect automatically
+   * 2. If user is not connected to any wallet:  On reload, connect to burner wallet
+   */
+  autoConnect: boolean;
+};
+
+const walletIdStorageKey = "scaffoldEth2.wallet";
+
+/**
+ * This function will get the initial connector (if any), the app will connect to
+ * @param config
+ * @param previousWalletId
+ * @param connectors
+ * @returns
+ */
+const getInitialConnector = (
+  config: TAutoConnect,
+  previousWalletId: string,
+  connectors: Connector<any, any, any>[],
+): { connector: Connector | undefined; chainId?: number } | undefined => {
+  const allowBurner = config.enableBurnerWallet;
+
+  if (!!!previousWalletId) {
```

Comment:
I think using only 1 bang should be enough :D

Reply:
actually you need 3.  !!! for a string means true if null, undefined or empty